Adds the SoCo.set_relative_volume() method.

pwt opened the pull request:
This PR adds the `set_relative_volume()` method to the `SoCo` class.

pwt commented:
Note: I couldn't devise a meaningful unit test for this method. Suggestions welcome.

KennethNielsen commented on soco/core.py:
```diff
@@ -479,6 +480,30 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
+        of 100, the volume will be set to 100. If the adjustment causes the
+        volume to undershoot the minimum value of 0, the volume will be set
+        to 0.
+
```
The SoCo instance has a `volume` property, which means that for relative changes you can do something like `device.volume += 10`, which has the same result as this BUT which requires 2 network calls. I think the documentation here should mention that option and the difference. You should probably check the other doc strings to see by what name a SoCo instance is usually referred to.

pwt replied:
Thanks for the comments. I've added documentation to refer to the increment/decrement assignment operators.

pwt commented:
Note that I've removed the bounds check in the method, partly because this is not meaningful without knowing the current volume, and partly because Sonos handles it  at the receiving end and does the right thing.

Anyway, I've added a unit test patterned on other tests. However, since it won't actually catch any errors, I'm not sure of its value. Perhaps I'm missing something.

pwt commented:
I'm hoping this is ready to merge now, once a committer gets a chance.

KennethNielsen commented on soco/core.py:
```diff
@@ -479,6 +480,36 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
+        Returns:
+            int: The new volume setting.
+        """
+        relative_volume = int(relative_volume)
```
Ok. So although this is hardly the standard for all code in soco (or even in Python in general), we might as well look to improve a little on the existing standard. In the Sphinx documentation-from-docstrings mechanics there is also the support for documenting exceptions that might be raised. I think this is a prime candidate, since the int call will fail with a `ValueError` if whatever you pass in cannot be converted to an int. So, as a last change I will suggest adding to the docstring, between the Args and the Returns, two lines like this:

```python
Raises:
    ValueError: If `relative_volume` cannot be cast as an integer
```

pwt commented:
I've made the docstring addition. Thanks for the comments on the unit tests. I'm starting to get a handle on how they're used in SoCo.

KennethNielsen commented:
Merged. Thanks for the contribution. Please add a comment to the release notes thread #674 about the change.
